src/tree-selector/treeSelectorComponent.ts

```diff
@@ -0,0 +1,69 @@
+import * as ng from 'angular';
+
+export class TreeSelectorController {
+
+  public initialData: any;
+  public selectables: any;
+  public openTreeView: any;
+  private reg: any;
+  private url: string;
+  private handleSelected: any;
+
+  /*@ngInject*/
+  constructor(private $http, private $timeout : ng.ITimeoutService){}
+
+  public $onInit() {
+    this.initialData = this.parseSelectable(this.initialData);
+    console.log('selectables: ', this.selectables, this.initialData);
+  }
+
+  private getData = (id?) => {
+    return this.$http({
```

Comment:
Are you 100% sure you want to use `$http` here? I'd move it out to the demo controller to make this more generic and so consumers can use their own service for fetching data. (Maybe I'm totally wrong, @himdel?)